File: 3251-maximum-area-of-longest-diagonal-rectangle/3251-maximum-area-of-longest-diagonal-rectangle.py
class Solution:
    def areaOfMaxDiagonal(self, dimensions: List[List[int]]) -> int:
        maxArea = 0
        maxDiag = 0
        for l,w in dimensions:
            diag = sqrt(l**2 + w**2)
            if diag > maxDiag:
                maxArea = l * w
                maxDiag = diag
            elif diag == maxDiag:
                maxArea = max(maxArea, l * w)
        return maxArea

File: 3251-maximum-area-of-longest-diagonal-rectangle/3251-maximum-area-of-longest-diagonal-rectangle.cpp
class Solution {
public:
    int areaOfMaxDiagonal(vector<vector<int>>& dimensions) {
        int maxArea = 0;
        double maxDiag = 0;
        for(int i = 0; i < dimensions.size(); i++)
        {
            double diag = sqrt(pow(dimensions[i][0], 2) + pow(dimensions[i][1], 2));
            if(diag > maxDiag)
            {
                maxArea = dimensions[i][0] * dimensions[i][1];
            }
            else if(diag == maxDiag)
            {
                maxArea = max(maxArea, dimensions[i][0] * dimensions[i][1]);
            }
            maxDiag = max(maxDiag, diag);
        }
        return maxArea;
    }
};